contracts/Nitro.sol

```diff
@@ -8,6 +8,141 @@ contract Nitro {
+     * @param initialGuaranteeOutcome the outcome containing at least one guarantee(s) which will be claimed for each asset.
+     * @param initialHoldings initial quantity of each asset held on chain for the guarantor channel. Order matches that of initialGuaranteeOutcome.
+     * @param targetChannel the index of the guarantee in the list of guarantees for the given asset -- equivalent to declaring a target channel
+     * @param initialTargetOutcome initial outcome stored on chain for the target channel.
+     * @param exitRequest list  of indices expressing which destinations in the allocation should be paid out for each asset.
+     */
+    function claim(
+        ExitFormat.SingleAssetExit[] memory initialGuaranteeOutcome,
+        uint256[] memory initialHoldings,
+        uint48 targetChannel,
+        ExitFormat.SingleAssetExit[] memory initialTargetOutcome,
+        uint48[][] memory exitRequest
+    )
+        public
+        pure
+        returns (
+            ExitFormat.SingleAssetExit[] memory updatedTargetOutcome,
+            uint256[] memory updatedHoldings,
+            ExitFormat.SingleAssetExit[] memory exit
+        )
+    {
+        require(initialTargetOutcome.length == initialHoldings.length);
+        require(initialTargetOutcome.length == initialGuaranteeOutcome.length);
+
+        exit = new ExitFormat.SingleAssetExit[](initialTargetOutcome.length);
+
+        // Iterate through every asset
+        for (
+            uint256 assetIndex = 0;
+            assetIndex < initialGuaranteeOutcome.length;
+            assetIndex++
+        ) {
+            ExitFormat.Allocation[] memory guarantees =
+                initialGuaranteeOutcome[assetIndex].allocations;
+
+            ExitFormat.Allocation[] memory targetAllocations =
+                initialTargetOutcome[assetIndex].allocations;
+
+            // If exitRequest is empty for the allocation we want ALL to exit
+            ExitFormat.Allocation[] memory exitAllocations =
+                new ExitFormat.Allocation[](
+                    exitRequest[assetIndex].length > 0
+                        ? exitRequest[assetIndex].length
+                        : targetAllocations.length
+                );
+
+            updatedTargetOutcome = new ExitFormat.SingleAssetExit[](
+                initialTargetOutcome.length
+            );
+            updatedHoldings = initialHoldings;
+
+            uint256 surplus = initialHoldings[assetIndex];
+            uint48 exitRequestIndex = 0;
+
+            require(
+                guarantees[targetChannel].callTo ==
+                    0x0000000000000000000000000000000000000001,
+                "Must be a valid guarantee with callTo set to MAGIC_VALUE_DENOTING_A_GUARANTEE"
+            );
+
+            address[] memory destinations =
+                ExitFormat.decodeGuaranteeData(guarantees[targetChannel].data);
```

Comment:
Is this the correct use of `targetChannel` ?

Reply:
`targetChannelIndex` might be a better name?

Reply:
173213e